infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingApplier.java

```diff
@@ -84,18 +85,19 @@ public void apply(InternalEnvironment internalEnvironment, Collection<ChePlugin>
 
     KubernetesEnvironment kubernetesEnvironment = (KubernetesEnvironment) internalEnvironment;
 
-    Map<String, Pod> pods = kubernetesEnvironment.getPods();
+    Map<String, PodData> pods = kubernetesEnvironment.getPodsData();
     switch (pods.size()) {
       case 0:
         addToolingPod(kubernetesEnvironment);
+        pods = kubernetesEnvironment.getPodsData();
```

Comment:
Can you elaborate on why we need to reassign the variable here?

Reply:
`getPodsData()` (and the other methods) have been changed to return immutable copies, so adding the tooling pod won't change `pods`.

Reply:
ok